packages/features/ee/teams/pages/team-members-view.tsx

```diff
@@ -25,6 +25,18 @@ const MembersView = () => {
     onError: () => {
       router.push("/settings");
     },
+    enabled: !!router.isReady,
+  });
+
+  const {
+    data: teamMembers,
+    isLoading: memberLoading,
+    hasNextPage,
+    fetchNextPage,
+    isFetchingNextPage,
+  } = trpc.useInfiniteQuery(["viewer.teams.getMembers", { teamId: Number(router.query.id), limit: 5 }], {
+    getNextPageParam: (lastPage) => lastPage.nextCursor,
+    enabled: !!router.isReady,
```

Comment:
The problem with this approach is that router.query.id isn't a number(we are calming typescript by doing an explicit typecast to a number). So, we aren't getting compile time safety. If we use zod over query it would fail.

One of the approaches that I use is to, make a wrapper component that renders this component only when router.isReady is true. After that in this component we should parse the schema using zod, because we now know that it has to be a parseable number.

Reply:
cc @zomars for review

Reply:
@hariombalhara can you point me to a file where you used the wrapper component?

Reply:
<img width="1056" alt="Screenshot 2022-10-17 at 9 09 05 PM" src="https://user-images.githubusercontent.com/1780212/196221425-3bc8b274-9800-42c1-aad8-43fb651df1e8.png">
Here is a similar approach, this approach solves the problem of making sure useForm defaultValues has non null value.

Similar approach can be used for this usecase.